### PR TITLE
Add OpenTelemetry sample and update deps

### PR DESCRIPTION
**Pull request by torosent**
This pull request updates several package dependencies in the `Directory.Packages.props` file to newer versions. The changes mainly focus on upgrading Microsoft.Extensions, gRPC/Protobuf, and other supporting libraries to improve compatibility, security, and stability.

# Summary
## What changed?
Dependency upgrades:

* Microsoft.Extensions.* packages: Upgraded all related packages (such as `Microsoft.Extensions.Caching.Memory`, `Microsoft.Extensions.DependencyInjection`, `Microsoft.Extensions.Hosting`, etc.) from version 10.0.x to 10.0.2 for consistency and potential bug fixes.
* gRPC and Protobuf packages: Updated `Google.Protobuf` to 3.33.5, `Grpc.Net.Client` to 2.76.0, `Grpc.Tools` to 2.78.0, and `Grpc.AspNetCore.Server` to 2.76.0, ensuring use of the latest stable versions.

Supporting library updates:

* `Microsoft.Bcl.AsyncInterfaces` and `System.Text.Json`: Upgraded to 10.0.2 for improved async support and JSON serialization.

Testing and analysis packages:

* Removed duplicate `BenchmarkDotNet` entry and ensured version 0.15.8 is used.…ing sample

- Updated Google.Protobuf package version from 3.33.2 to 3.33.5 in AzureFunctionsApp and Worker.Grpc projects.
- Removed outdated package references in Directory.Packages.props for Functions sample.
- Added OpenTelemetry packages for distributed tracing in Directory.Packages.props.
- Introduced a new DistributedTracingSample project demonstrating OpenTelemetry integration with Durable Task SDK.
- Implemented a fan-out/fan-in orchestration pattern in the DistributedTracingSample.
- Created a README.md for the DistributedTracingSample with setup instructions and usage details.
- Added a docker-compose.yml for running Jaeger and the DTS emulator for testing the sample.

## Why is this change needed?
- Adding sample

## Issues / work items
- Resolves #
- Related #

---

# Project checklist
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [x] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): Copilot CLI - Opus 4.6
- AI-assisted areas/files: All

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed

## Manual validation (only if runtime/behavior changed)
- Environment (OS, .NET version, components): OSX .NET 10

- Evidence (optional):
image
image

---

# Notes for reviewers
- N/A

### CODE REVIEW

**Comment by torosent** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/microsoft/durabletask-dotnet/pull/637#pullrequestreview-3766153431)

File: samples/DistributedTracingSample/Program.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+// Schedule the orchestration and wait for it to complete.
+await using DurableTaskClient client = host.Services.GetRequiredService<DurableTaskClient>();
+string instanceId = await client.ScheduleNewOrchestrationInstanceAsync("FanOutFanIn");
+Console.WriteLine($"Started orchestration instance: '{instanceId}'");
+
+using CancellationTokenSource timeoutCts = new CancellationTokenSource(TimeSpan.FromSeconds(60));
+OrchestrationMetadata result = await client.WaitForInstanceCompletionAsync(
+    instanceId,
+    getInputsAndOutputs: true,
```
[the remark this answers is not in the document]
Comment (torosent) [reply samples/DistributedTracingSample/Program.cs:94]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/microsoft/durabletask-dotnet/pull/637#discussion_r2776934279)

**Review by YunchuWang — APPROVED** [review]
